opentracing-spring-rabbitmq-it/src/test/java/io/opentracing/contrib/spring/rabbitmq/RabbitMqSendAndReceiveTracingItTest.java

```diff
@@ -0,0 +1,262 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.amqp.core.BindingBuilder;
+import org.springframework.amqp.core.Queue;
+import org.springframework.amqp.core.TopicExchange;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.connection.RabbitConnectionFactoryBean;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer;
+import org.springframework.amqp.rabbit.listener.adapter.MessageListenerAdapter;
+import org.springframework.amqp.support.converter.SimpleMessageConverter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.Assert;
+
+/**
+ *  @author Gilles Robert
+ */
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = {RabbitMqSendAndReceiveTracingItTest.TestConfig.class,
+    RabbitMqSendAndReceiveTracingItTest.RabbitListenerTestConfig.class}
+)
+@RunWith(SpringJUnit4ClassRunner.class)
+public class RabbitMqSendAndReceiveTracingItTest {
+
+  private final EmbeddedQpidBroker broker = new EmbeddedQpidBroker();
+  @Autowired private RabbitTemplate rabbitTemplate;
+  @Autowired private MockTracer tracer;
+  @Autowired private RabbitConnectionFactoryBean rabbitConnectionFactoryBean;
+  @Autowired private Queue queue;
+
+  @Before
+  public void setup() throws Exception {
+    tracer.reset();
+    broker.start();
+
+    final CachingConnectionFactory cachingConnectionFactory =
+        new CachingConnectionFactory(rabbitConnectionFactoryBean.getObject());
+
+    final TopicExchange exchange = new TopicExchange("myExchange", true, false);
+
+    final RabbitAdmin admin = new RabbitAdmin(cachingConnectionFactory);
+    admin.declareQueue(queue);
+    admin.declareExchange(exchange);
+    admin.declareBinding(BindingBuilder.bind(queue).to(exchange).with("#"));
+
+    cachingConnectionFactory.destroy();
+  }
+
+  @After
+  public void deleteExchange() throws Exception {
+    final CachingConnectionFactory cachingConnectionFactory =
+        new CachingConnectionFactory(rabbitConnectionFactoryBean.getObject());
+    final RabbitAdmin admin = new RabbitAdmin(cachingConnectionFactory);
+    admin.deleteExchange("myExchange");
+    cachingConnectionFactory.destroy();
+    broker.stop();
+  }
+
+  @Test
+  public void testSendAndReceiveRabbitMessage() {
+    final String message = "hello world message!";
+    rabbitTemplate.convertAndSend("myExchange", "#", message);
```

Comment:
I would like to see a similar test when there is an active span present. The test would verify that parent span of the rabbit span is correct.

Reply:
It can be also done here.